clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -744,7 +744,7 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
             do {
 
                 // Make sure to let the background thread know that we are still polling.
-                applicationEventHandler.add(new PollEvent(timer.currentTimeMs()));
+                applicationEventHandler.add(new PollEvent(timer.currentTimeMs(), subscriptions.allConsumed()));
```

Comment:
We've done a lot of work to ensure the background thread "owns" the current subscription state. This seems to go against those efforts.

Can we add a comment to this line that explains why we don't want to the background thread to "own" the set of consumed partitions? This would be helpful since later on during the poll process we _do_ implicitly let the background thread determine the partitions to fetch (rather than having the application thread pass in the partitions).